src/Shelly1/shelly_init.cpp

```diff
@@ -54,10 +64,29 @@ void CreateComponents(std::vector<std::unique_ptr<Component>> *comps,
     comps->emplace_back(std::move(gdo));
     return;
   }
-  // Single switch with non-detached input = only one accessory.
-  bool to_pri_acc = (mgos_sys_config_get_sw1_in_mode() != 3);
+
+  // Sensor Discovery
+  std::unique_ptr<OWSensorManager> manager(new OWSensorManager(onewire.get()));
```

Comment:
this object lives only until the function returns.
why is OWSensorManager a class at all if we ever call only one of its methods?
this smells of java and i don't like that smell :)

Reply:
yes, is now a method of onewire